jupyterlab/labapp.py

```diff
@@ -258,6 +260,91 @@ def start(self):
         print('Workspaces directory: %s' % get_workspaces_dir())
 
 
+class LabWorkspaceListApp(JupyterApp):
+    version = version
+    description = """
+    Print all the workspaces available
+
+    If '--json' flag is passed in, 'json' output of each workspace is printed.
+    If '--jsonlist' flag is passed in, 'json' output of list of workspaces is printed.
+    If nothing is passed in, 'human-readable' output is printed.
+    """
+    flags = dict(
+        jsonlist=(
+            {"LabWorkspaceListApp": {"jsonlist": True}},
+            _i18n("Produce machine-readable JSON list output."),
+        ),
+        json=(
+            {"LabWorkspaceListApp": {"json": True}},
+            _i18n("Produce machine-readable JSON object on each line of output."),
+        ),
+    )
+
+    jsonlist = Bool(
+        False,
+        config=True,
+        help=_i18n(
+            "If True, the output will be a JSON list of objects, one per "
+            "JupyterLab workspace, each with the details of the "
+            "relevant workspace"
+        ),
+    )
+    json = Bool(
+        False,
+        config=True,
+        help=_i18n(
+            "If True, each line of output will be a JSON object with the "
+            "details of the workspace."
+        ),
+    )
+
+    def start(self):
+        app = LabApp(config=self.config)
+        directory = app.workspaces_dir
+
+        items = [item
+                for item in os.listdir(directory)
+                if item.endswith(WORKSPACE_EXTENSION)]
+        items.sort()
+
+        workspaces = _list_available_workspaces(directory, items)
+        if self.jsonlist:
+            print(json.dumps(workspaces))
```

Comment:
would want to use `encoding='utf-8'`, probably

Reply:
It does not seem to be supported: https://docs.python.org/3/library/json.html#json.dumps